x-pack/plugins/rollup/public/crud_app/sections/components/job_action_menu/job_action_menu.js

```diff
@@ -0,0 +1,287 @@
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      isPopoverOpen: false,
+      showDeleteConfirmation: false
+    };
+  }
+
+  panels() {
+    const {
+      startJobs,
+      stopJobs,
+      intl,
+    } = this.props;
+
+    const isSingleSelection = this.isSingleSelection();
+    const entity = this.getEntity(isSingleSelection);
+
+    const items = [];
+
+    if (this.canStartJobs()) {
+      items.push({
+        name: intl.formatMessage({
+          id: 'xpack.rollupJobs.jobActionMenu.startJobLabel',
+          defaultMessage: 'Start {entity}',
+        }, { entity }),
+        icon: <EuiIcon type="play" />,
+        onClick: () => {
+          this.closePopover();
+          startJobs();
+        },
+      });
+    }
+
+    if (this.canStopJobs()) {
+      items.push({
+        name: intl.formatMessage({
+          id: 'xpack.rollupJobs.jobActionMenu.stopJobLabel',
+          defaultMessage: 'Stop {entity}',
+        }, { entity }),
+        icon: <EuiIcon type="pause" />,
+        onClick: () => {
+          this.closePopover();
+          stopJobs();
+        },
+      });
+    }
+
+    items.push({
+      name: intl.formatMessage({
+        id: 'xpack.rollupJobs.jobActionMenu.deleteJobLabel',
+        defaultMessage: 'Delete {entity}',
+      }, { entity }),
+      icon: <EuiIcon type="trash" />,
+      onClick: () => {
+        this.closePopover();
+        this.openDeleteConfirmationModal();
+      },
+    });
+
+    const upperCasedEntity = `${entity[0].toUpperCase()}${entity.slice(1)}`;
+    const panelTree = {
+      id: 0,
+      title: intl.formatMessage({
+        id: 'xpack.rollupJobs.jobActionMenu.panelTitle',
+        defaultMessage: '{upperCasedEntity} options',
+      }, { upperCasedEntity }),
+      items,
+    };
+
+    return flattenPanelTree(panelTree);
+  }
+
+  onButtonClick = () => {
+    this.setState(prevState => ({
+      isPopoverOpen: !prevState.isPopoverOpen
+    }));
+  };
+
+  closePopover = () => {
+    this.setState({
+      isPopoverOpen: false
+    });
+  };
+
+  closeDeleteConfirmationModal = () => {
+    this.setState({ showDeleteConfirmation: false });
+  };
+
+  openDeleteConfirmationModal = () => {
+    this.setState({ showDeleteConfirmation: true });
+  };
+
+  canStartJobs() {
+    const { jobs } = this.props;
+    return jobs.some(job => job.status === 'stopped');
+  }
+
+  canStopJobs() {
+    const { jobs } = this.props;
+    return jobs.some(job => job.status === 'started');
+  }
+
+  confirmDeleteModal = () => {
+    const isSingleSelection = this.isSingleSelection();
+    const entity = this.getEntity(isSingleSelection);
+
+    const {
+      deleteJobs,
+      jobs,
+      intl,
+    } = this.props;
+
+    const { showDeleteConfirmation } = this.state;
+
+    if (!showDeleteConfirmation) {
+      return null;
+    }
+
+    const onDelete = () => {
+      this.closePopover();
+      deleteJobs();
+    };
+
+    return (
+      <EuiOverlayMask>
+        <EuiConfirmModal
```

Comment:
![image](https://user-images.githubusercontent.com/1965714/44491267-8c1c7780-a615-11e8-8e18-520493f31987.png)
![image](https://user-images.githubusercontent.com/1965714/44491298-a8201900-a615-11e8-9925-0223baec5d5a.png)

we should follow [EUI guidelines](https://elastic.github.io/eui/#/guidelines/modals) for this Delete modal more closely:
- for deleting one job, have the job ID be present in title
- for more than one job, have job count in title
- danger color for Delete button
- no need for callout

also, I wonder if it would be helpful to doubly warn the user if they are deleting job(s) that are already started? it's not an issue server-side - ES seems to stop the job before deletion, so this would be purely UX to ensure that the user doesn't delete a running job they may have selected by accident.

Reply:
Good catch! Done.

![image](https://user-images.githubusercontent.com/1238659/44504798-bd676880-a652-11e8-909f-012bbd860fb3.png)

![image](https://user-images.githubusercontent.com/1238659/44504800-be989580-a652-11e8-8bcd-622f5cc01a23.png)

![image](https://user-images.githubusercontent.com/1238659/44504802-c0625900-a652-11e8-96ae-fbc858179509.png)